internal/handlers/common/distinct.go

```diff
@@ -49,6 +52,22 @@ func GetDistinctParams(document *types.Document, l *zap.Logger) (*DistinctParams
 		return nil, err
 	}
 
+	switch filter := dp.Query.(type) {
+	case *types.Document:
+		dp.Filter = filter
+	case types.NullType, nil:
+		dp.Filter = types.MakeDocument(0)
+	default:
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrTypeMismatch,
+			fmt.Sprintf(
+				"BSON field 'distinct.query' is the wrong type '%s', expected type 'object'",
+				commonparams.AliasFromType(dp.Query),
+			),
+			"distinct",
+		)
```

Comment:
Could you add test to check the error message please?

Reply:
Done.